lib/cocoapods/installer/project_cache/project_installation_cache.rb

```diff
@@ -0,0 +1,66 @@
+module Pod
+  class Installer
+    module ProjectCache
+      # Represents the cache stored at Pods/.project/installation_cache
+      #
+      class ProjectInstallationCache
+        require 'cocoapods/installer/project_cache/target_cache_key'
+
+        # @return [Hash{String => TargetCacheKey}]
+        #         Stored hash of target cache key objects for every pod target.
+        #
+        attr_reader :cache_key_by_target_label
+
+        # @return [Hash{String => Symbol}]
+        #         Build configurations stored in the cache.
+        #
+        attr_reader :build_configurations
+
+        # @return [String]
+        #         Project object stored in the cache.
+        #
+        attr_reader :project_object_version
+
+        # Initializes a new instance.
+        #
+        # @param [Hash{String => TargetCacheKey}] cache_key_by_target_label @see #cache_key_by_target_label
+        # @param [Hash{String => Symbol}] build_configurations @see #build_configurations
+        # @param [String] project_object_version @see #project_object_version
+        #
+        def initialize(cache_key_by_target_label = {}, build_configurations = nil, project_object_version = nil)
+          @cache_key_by_target_label = cache_key_by_target_label
+          @build_configurations = build_configurations
+          @project_object_version = project_object_version
+        end
+
+        attr_writer :cache_key_by_target_label
+
+        attr_writer :build_configurations
+
+        attr_writer :project_object_version
+
+        def save_as(path)
+          Pathname(path).dirname.mkpath
+          cache_key_contents = Hash[cache_key_by_target_label.map do |label, key|
+            [label, key.to_h]
+          end]
+          contents = { 'CACHE_KEYS' => cache_key_contents }
+          contents['BUILD_CONFIGURATIONS'] = build_configurations if build_configurations
+          contents['OBJECT_VERSION'] = project_object_version if project_object_version
```

Comment:
could the creation of `contents` be pulled out into something similar to `to_hash` in `ProjectMetadataCache`? makes it easier to test without having to write to a file

Reply:
Good idea. Will do